### PR TITLE
feat(button): migrate button component

### PR DESCRIPTION
**Pull request by mJarsater**
**Describe pull-request**  
Migration of the button component, both native and webcomponent

**Solving issue**  
Fixes: [AB#2398](https://dev.azure.com/scaniadigitaldesignsystem/3d9eb9c6-0045-473d-bc1d-e842dd779200/_workitems/edit/2398)

**How to test**  
_Add description how to test if possible_
1. Go to storybook link below
2. Check in components -> button
3. Check that the design is aligned with the one in Figma.

**Screenshots**  
-

**Additional context**  
-

### CODE REVIEW

File: components/src/components/button/button.stories.js [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+      control: {
+        type: 'radio',
+        options: ['primary', 'secondary', 'ghost', 'danger'],
+      },
+    },
+    size: {
+      name: 'Size',
+      control: {
+        type: 'radio',
+        options: ['sm', 'md'],
```
[the remark this answers is not in the document]
Comment (EmelieLitwin) [reply components/src/components/button/button.stories.js:25]: This is something we need to decide on so we are consistent. I don't always use the prop names and options (for example I use 'small', 'medium', etc. There are also some other props that I feel are a little unclear) and count on that they see it in the code examples.

[the remark this answers is not in the document]
Comment (mJarsater) [reply components/src/components/button/button.stories.js:25]: @EmelieLitwin totally agree. Maybe something we can discuss at the devCheckin?

File: components/src/components/button/button.stories.js [hunk 2]
```diff
@@ -0,0 +1,158 @@
+/* Default controls */
+
+export default {
+  title: 'Component/Button',
```
Comment (victorscania) [inline components/src/components/button/button.stories.js:4]: Should this not be Components/Button. or are we doing for singular? "Component"

Comment (mJarsater) [reply components/src/components/button/button.stories.js:4]: No like you said it should be Components, I'll fix it 👍🏻

**Review by victorscania — COMMENTED** [review]
Native folder for the native stories?
Screenshot 2022-10-20 at 15 14 51

**Comment by theJohnnyMe** [comment]
Wait with this one, you will need some updates from the main in order to continue.

**Pull request closed without merging**